### PR TITLE
Migrate eventgrid tests

### PR DESCRIPTION
**Pull request by g2vinay**
Migrate Eventgrid tests

### CODE REVIEW

**Comment by g2vinay** [comment]
To Do: Re-record tests

File: sdk/eventgrid/test-resources.json [hunk 1]
```diff
@@ -222,6 +215,26 @@
         "scope": "[resourceGroup().id]"
       }
     },
+    {
```
Comment (billwert) [inline sdk/eventgrid/test-resources.json:218]: why are we duplicating these roles?

File: sdk/eventgrid/azure-messaging-eventgrid/src/test/java/com/azure/messaging/eventgrid/EventGridPublisherClientTests.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ protected void beforeTest() {
     public void publishEventGridEvents() {
         EventGridPublisherAsyncClient<EventGridEvent> egClient = builder
             .endpoint(getEndpoint(EVENTGRID_ENDPOINT))
-            .credential(getKey(EVENTGRID_KEY))
+            .credential(TestUtil.getTestTokenCredential(interceptorManager))
```
Comment (billwert) [inline sdk/eventgrid/azure-messaging-eventgrid/src/test/java/com/azure/messaging/eventgrid/EventGridPublisherClientTests.java:78]: Is it permitted to leave the key-based tests as live-only? Switching to bearer auth means we're missing on the `AzureKeyCredential` coverage.

File: sdk/eventgrid/azure-messaging-eventgrid-cloudnative-cloudevents/pom.xml [hunk 3]
```diff
@@ -90,6 +90,11 @@
       <artifactId>azure-messaging-eventgrid</artifactId>
       <version>4.23.0-beta.1</version> <!-- {x-version-update;com.azure:azure-messaging-eventgrid;current} -->
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-identity</artifactId>
+      <version>1.13.0</version> <!-- {x-version-update;com.azure:azure-identity;dependency} -->
```
Comment (g2vinay) [inline sdk/eventgrid/azure-messaging-eventgrid-cloudnative-cloudevents/pom.xml:96]: this should scope to test only.

Comment (billwert) [reply sdk/eventgrid/azure-messaging-eventgrid-cloudnative-cloudevents/pom.xml:96]: good catch, fixed.

**Review by billwert — APPROVED** [review]

**Pull request merged**